src/components/RegistrationForm.jsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { Form, Button, Icon, Modal, Message } from "semantic-ui-react";
+import { useDispatch, useSelector } from "react-redux";
+import { signUp } from "../modules/Authentication";
+
+const RegistrationForm = () => {
+  const { registerErrorMessage, modalOpen } = useSelector(state => state);
+  const dispatch = useDispatch();
+
+  return (
+    <>
+      <Modal
+        closeIcon
+        onClose={() => dispatch({ type: "CLOSE_MODAL" })}
+        onOpen={() => dispatch({ type: "OPEN_MODAL" })}
```

Comment:
not really a problem a.t.m but when we add the `subscriber modal` we might want to change this name "CLOSE_REGISTATION" and "OPEN_REGISTRATION"

Reply:
Changed them to OPEN_REGISTRATION_FORM and CLOSE_REGISTRATION_FORM